cli/src/main/java/com/devonfw/tools/ide/property/Property.java

```diff
@@ -170,51 +178,199 @@ public String getValueAsString() {
 
-  @Override
-  public int hashCode() {
+  /**
+   * @param args the {@link CliArguments} already {@link CliArguments#current() pointing} the {@link CliArgument} to
+   *        apply.
+   * @param context the {@link IdeContext}.
+   * @param commandlet the {@link Commandlet} owning this property.
+   * @param collector the {@link CompletionCandidateCollector}.
+   * @return {@code true} if it matches, {@code false} otherwise.
+   */
+  public boolean apply(CliArguments args, IdeContext context, Commandlet commandlet,
+      CompletionCandidateCollector collector) {
 
-    return Objects.hash(this.name, this.value);
+    CliArgument argument = args.current();
+    if (argument.isCompletion()) {
+      return complete(argument.get(), args, context, commandlet, collector);
+    }
+    boolean option = isOption();
+    if (option && !argument.isOption()) {
+      return false;
+    }
+    String argValue = null;
+    boolean lookahead = false;
+    if (this.name.isEmpty()) {
+      argValue = argument.get();
+    } else {
+      if (!matches(argument.getKey())) {
+        return false;
+      }
+      argValue = argument.getValue();
+      if (argValue == null) {
+        argument = args.next();
+        if (argument.isCompletion()) {
+          return completeValue(argument.get(), context, commandlet, collector);
+        } else {
+          if (!argument.isEnd()) {
+            argValue = argument.get();
+          }
+          lookahead = true;
+        }
+      }
+    }
+    return applyValue(argValue, lookahead, args, context, commandlet, collector);
+  }
+
+  /**
+   * @param argValue the value to set as {@link String}.
+   * @param lookahead - {@code true} if the given {@code argValue} is taken as lookahead from the next value,
+   *        {@code false} otherwise.
+   * @param args the {@link CliArguments}.
+   * @param context the {@link IdeContext}.
+   * @param commandlet the {@link Commandlet} owning this {@link Property}.
+   * @param collector the {@link CompletionCandidateCollector}.
+   * @return {@code true} if it matches, {@code false} otherwise.
+   */
+  protected boolean applyValue(String argValue, boolean lookahead, CliArguments args, IdeContext context,
+      Commandlet commandlet, CompletionCandidateCollector collector) {
+
+    boolean success = assignValueAsString(argValue, context, commandlet);
+    if (success) {
+      args.next();
+    }
+    return success;
+  }
+
+  /**
+   * Performs auto-completion for the {@code arg}.
+   *
+   * @param arg the {@link CliArgument#get() CLI argument}.
+   * @param args TODO
+   * @param context the {@link IdeContext}.
+   * @param commandlet the {@link Commandlet} owning this {@link Property}.
+   * @param collector the {@link CompletionCandidateCollector}.
+   * @return {@code true} if it matches, {@code false} otherwise.
+   */
+  protected boolean complete(String arg, CliArguments args, IdeContext context, Commandlet commandlet,
```

Comment:
Lots of TODO's. Maybe add an issue for each TODO and link it here?

Reply:
I need ~2 h to get this done, but too much other work and too little time for IDE at the moment. Will try to get it done by end of the week.

Reply:
Was already solved in PR #187 and is merged. Not part of this PR anymore. Please resolve.